File: python/get_branches.py
import requests

def get_branches(github_base_url :str, user_name :str, repository :dict, headers :dict) -> tuple[dict, str]:
    github_branches_url = f"{github_base_url}/repos/{user_name}/{repository['name']}/branches"
    response = requests.request("GET", github_branches_url, headers=headers)

    if response.status_code == 200:
        branches = response.json()
        return branches, ""
    else:
        return {}, f"Failed to retrieve branches. Status code: {response.status_code}"

File: go/get_branches.go
package main

import (
	"encoding/json"
	"fmt"
	"io"
	"log"
	"net/http"
)

func get_branches(client *http.Client, Bearer string, repository Repository) []Branch {

	fetch_branches_url := fmt.Sprintf("%s/repos/%s/%s/branches", GITHUB_BASE_URL, USER_NAME, repository.Name)
	req, err := http.NewRequest("GET", fetch_branches_url, nil)
	if err != nil {
		log.Fatalln(err)
	}
	req.Header.Set("Authorization", Bearer)

	resp, err := client.Do(req)
	if err != nil {
		log.Println("Error on response.\n[ERROR] -", err)
	}
	defer resp.Body.Close()

	body, err := io.ReadAll(resp.Body)
	if err != nil {
		log.Println("Error while reading the response bytes:", err)
	}

	var branches []Branch

	err = json.Unmarshal(body, &branches)
	if err != nil {
		log.Println("Error while marshalling the response bytes:", err)
	}

	return branches
}
